modules/homa-lite/src/lib.rs

```diff
@@ -488,74 +492,42 @@ pub mod module {
 				// Deduct BaseWithdrawFee from the liquid amount.
 				let liquid_amount = liquid_amount.saturating_sub(base_withdraw_fee);
 
-				// If there are available_staking_balances, redeem immediately with no additional fee.
-				let available_staking_balance = Self::available_staking_balance();
-				let actual_liquid_amount = min(
-					liquid_amount,
-					Self::convert_staking_to_liquid(available_staking_balance)?,
-				);
-
-				let mut liquid_remaining = liquid_amount;
-				if Self::convert_liquid_to_staking(actual_liquid_amount)? > T::XcmUnbondFee::get() {
-					// Immediately redeem from the available_staking_balances
-					let actual_staking_amount = Self::convert_liquid_to_staking(actual_liquid_amount)?;
-
-					// Redeem from the available_staking_balances costs additional fee for Xcm unbond.
-					T::Currency::deposit(
-						T::StakingCurrencyId::get(),
+				// Reserve/unreserve the difference amount.
+				match liquid_amount.cmp(&old_amount) {
+					// Lock more liquid currency.
+					Ordering::Greater => T::Currency::reserve(
+						T::LiquidCurrencyId::get(),
 						&who,
-						actual_staking_amount.saturating_sub(T::XcmUnbondFee::get()),
-					)?;
-					let slash_amount = T::Currency::slash(T::LiquidCurrencyId::get(), &who, actual_liquid_amount);
-					ensure!(slash_amount.is_zero(), Error::<T>::InsufficientLiquidBalance);
-
-					// Update the available_staking_balance
-					let available_staking_balance = available_staking_balance.saturating_sub(actual_staking_amount);
-					AvailableStakingBalance::<T>::put(available_staking_balance);
-
-					Self::deposit_event(Event::<T>::Redeemed(
-						who.clone(),
-						actual_staking_amount,
-						actual_liquid_amount,
-					));
-					liquid_remaining = liquid_remaining.saturating_sub(actual_liquid_amount);
-				}
-
-				// Unredeemed requests are added to a queue.
-				if Self::liquid_amount_is_above_minimum_threshold(liquid_remaining) {
-					// Check if there's already a queued redeem request.
-					let (request_amount, _) = Self::redeem_requests(&who).unwrap_or((0, Permill::default()));
-
-					match liquid_remaining.cmp(&request_amount) {
-						// Lock more liquid currency.
-						Ordering::Greater => T::Currency::reserve(
+						liquid_amount.saturating_sub(old_amount),
+					),
```

Comment:
why not just error return here?
```suggestion
					)?,
```

Reply:
true. I will add more test to ensure this will fail when the user has insufficent fund